include/misc/sem.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * @brief public sys_sem APIs.
+ */
+
+#ifndef ZEPHYR_INCLUDE_MISC_SEM_H_
+#define ZEPHYR_INCLUDE_MISC_SEM_H_
+
+#ifdef CONFIG_USERSPACE
```

Comment:
You need an alternate implementation which maps directly onto k_sem APIs if userspace isn't enabled.

See what was done in include/misc/mutex.h